UIforETW/Utility.cpp

```diff
@@ -349,78 +558,83 @@ int DeleteFiles(HWND hwnd, const std::vector<std::wstring>& paths)
 		FO_DELETE,
 		&fileNames[0],
 		NULL,
-		FOF_ALLOWUNDO | FOF_FILESONLY | FOF_NOCONFIRMATION,
+		(FOF_ALLOWUNDO | FOF_FILESONLY | FOF_NOCONFIRMATION),
 	};
 	// Delete using the recycle bin.
-	int result = SHFileOperation(&fileOp);
-
-	return result;
+	//TODO: IFileOperation?
+	return ::SHFileOperationW(&fileOp);
 }
 
 void SetClipboardText(const std::wstring& text)
 {
-	BOOL cb = OpenClipboard(GetDesktopWindow());
-	if (!cb)
+	ATLVERIFY(::OpenClipboard(::GetDesktopWindow()));
+	ATLVERIFY(::EmptyClipboard());
+
+	const size_t length = (text.size() + 1) * sizeof(text[0]);
+	HANDLE hmem = ::GlobalAlloc(GMEM_MOVEABLE, length);
+	if (!hmem)
+	{
+		ATLVERIFY(::CloseClipboard());
 		return;
+	}
 
-	EmptyClipboard();
+	void* const ptr = ::GlobalLock(hmem);
+	UIETWASSERT(ptr != NULL);
 
-	size_t length = (text.size() + 1) * sizeof(wchar_t);
-	HANDLE hmem = GlobalAlloc(GMEM_MOVEABLE, length);
-	if (hmem)
-	{
-		void *ptr = GlobalLock(hmem);
-		if (ptr != NULL)
-		{
-			memcpy(ptr, text.c_str(), length);
-			GlobalUnlock(hmem);
+	wcscpy_s(static_cast<wchar_t*>(ptr), (text.size() + 1), text.c_str());
 
-			SetClipboardData(CF_UNICODETEXT, hmem);
-		}
+	UnlockGlobalMemory(hmem);
+	if (::SetClipboardData(CF_UNICODETEXT, hmem) == NULL)
+	{
+		ATLVERIFY(!::GlobalFree(hmem));
+		ATLVERIFY(::CloseClipboard());
+		return;
 	}
-
-	CloseClipboard();
+	ATLVERIFY(::CloseClipboard());
 }
 
 std::wstring GetClipboardText()
 {
+	ATLVERIFY(::OpenClipboard(::GetDesktopWindow()));
 	std::wstring result;
-	BOOL cb = OpenClipboard(GetDesktopWindow());
-	if (!cb)
-		return result;
 
-	HANDLE hClip = GetClipboardData(CF_UNICODETEXT);
-	if (hClip)
+	HANDLE hClip = ::GetClipboardData(CF_UNICODETEXT);

```

Comment:
Woah - GetClipboardData can *definitely* fail. There might not be any text in the clipboard - could be an image instead. Possible errors still need to be handled.